picard/tagger.py

```diff
@@ -291,6 +293,24 @@ def __init__(self, picard_args, unparsed_args, localedir, autoupdate):
         if self.autoupdate_enabled:
             self.updatecheckmanager = UpdateCheckManager(parent=self.window)
 
+        from queue import Queue
+        self._pending_files = {}
+        self._pending_files_process_queue = mp.Queue()
+        self._pending_files_results_queue = mp.Queue()
+        self._pending_load_finished_queue = Queue()
+        self._file_loaded_queue = Queue()
+        self._process_stopping = mp.Value('b', False)
+        self._loader_process = mp.Process(target=File._load_check_metadata,
+                                          args=(self._process_stopping,
+                                                self._pending_files_process_queue,
+                                                self._pending_files_results_queue))
```

Comment:
Perhaps we could move all those args to an object.
Something like

```python
class PendingFiles:

   def __init__():
        self.files = {}
        self.process_queue = mp.Queue()
        self.results_queue = mp.Queue()
        self.stopping = mp.Value('b', False)

    def stop():
        self.process_stopping.value = True
        self.process_queue.put(None)
        self.process_queue.cancel_join_thread()
 
    def add_files(new_files, target):
           for file in new_files:
                self.files[file.filename] = (file, target)
                self.process_queue.put(file.filename)


self._pending_files = PendingFiles()

self._loader_process = mp.Process(target=File._load_check_metadata, args=(self._pending_files, ))

```

Reply:
Tried it, but it can't be pickled to be sent as the process argument.